### PR TITLE
ITSASU-786: update redirect URL for agent uk property CYA

### PR DESCRIPTION
**Pull request by scullythornton1**

### CODE REVIEW

File: app/controllers/agent/business/PropertyAccountingMethodController.scala [hunk 1]
```diff
@@ -67,30 +72,39 @@ class PropertyAccountingMethodController @Inject()(propertyAccountingMethod: Pro
     implicit user =>
       withAgentReference { reference =>
         subscriptionDetailsService.fetchIncomeSource(reference) flatMap {
```
Comment (sb-dev) [inline app/controllers/agent/business/PropertyAccountingMethodController.scala:74]: It looks like you are fetching the income source twice. The code can change to:
```
      withAgentReference { reference =>
        AccountingMethodPropertyForm.accountingMethodPropertyForm.bindFromRequest.fold(
          formWithErrors => Future.successful(BadRequest(view(
            accountingMethodForm = formWithErrors,
            isEditMode = isEditMode
          ))),

          accountingMethodProperty => {
            subscriptionDetailsService.saveAccountingMethodProperty(reference, accountingMethodProperty) flatMap { _ =>
              if (isSaveAndRetrieve) {
                Future(Redirect(controllers.agent.business.routes.PropertyCheckYourAnswersController.show(isEditMode)))
              } else {
                subscriptionDetailsService.fetchIncomeSource(reference) map {
                  case Some(IncomeSourceModel(_, _, true)) =>
                    Redirect(controllers.agent.business.routes.OverseasPropertyStartDateController.show())
                  case _ =>
                    Redirect(controllers.agent.routes.CheckYourAnswersController.show)
                }
              }
            }
          }
        )
      }
```

File: app/controllers/agent/business/PropertyStartDateController.scala [hunk 2]
```diff
@@ -42,15 +45,17 @@ class PropertyStartDateController @Inject()(val propertyStartDate: PropertyStart
...
   def view(propertyStartDateForm: Form[DateModel], isEditMode: Boolean, incomeSourceModel: IncomeSourceModel)
           (implicit request: Request[_]): Html = {
     propertyStartDate(
       propertyStartDateForm = propertyStartDateForm,
       postAction = controllers.agent.business.routes.PropertyStartDateController.submit(editMode = isEditMode),
       isEditMode = isEditMode,
-      backUrl = backUrl(isEditMode, incomeSourceModel)
+      backUrl = backUrl(isEditMode, Some(incomeSourceModel))
     )
   }
```
Comment (sb-dev) [inline app/controllers/agent/business/PropertyStartDateController.scala:61]: Should we add a test for `case None => Redirect(controllers.agent.routes.IncomeSourceController.show())` in the `show` function?

Comment (scullythornton1) [reply app/controllers/agent/business/PropertyStartDateController.scala:61]: added the test

File: app/controllers/agent/business/PropertyAccountingMethodController.scala (hunk 1 above)
Comment (scullythornton1) [reply app/controllers/agent/business/PropertyAccountingMethodController.scala]: amended as suggested. thanks

**Review by sb-dev — APPROVED** [review]
